src/core/server/http/http_server.test.ts

```diff
@@ -578,6 +579,90 @@ test('exposes route details of incoming request to a route handler', async () =>
     });
 });
 
+describe('conditional compression', () => {
+  async function setupServer(innerConfig: HttpConfig) {
+    const { registerRouter, server: innerServer } = await server.setup(innerConfig);
+    const router = new Router('', logger, enhanceWithContext);
+    // we need the large body here so that compression would normally be used
+    const largeRequest = {
+      body: 'hello'.repeat(500),
+      headers: { 'Content-Type': 'text/html; charset=UTF-8' },
+    };
+    router.get({ path: '/', validate: false }, (_context, _req, res) => res.ok(largeRequest));
+    registerRouter(router);
+    await server.start();
+    return innerServer.listener;
+  }
```

Comment:
This looks like an integration test and should be in `src/core/server/http/integration_tests` instead

Reply:
Good call, I'll move it.

Reply:
As discussed offline, looks like these tests / usage of supertest are in the correct place after all =)